src/promptflow-evals/promptflow/evals/synthetic/simulator/simulator/simulator.py

```diff
@@ -0,0 +1,735 @@
+from promptflow.evals.synthetic.simulator.templates.simulator_templates import SimulatorTemplates, Template
+
+BASIC_MD = os.path.join(template_dir, "basic.md")  # type: ignore[has-type]
+USER_MD = os.path.join(template_dir, "user.md")  # type: ignore[has-type]
+
+
+class Simulator:
+    def __init__(
+        self,
+        simulator_connection: "AzureOpenAIModelConfiguration" = None,  # type: ignore[name-defined]
+        ai_client: "AIClient" = None,  # type: ignore[name-defined]
+        ml_client: "MLClient" = None,  # type: ignore[name-defined]
+        simulate_callback: Optional[Callable[[Dict], Dict]] = None,
+    ):
+        """
+        Initialize the instance with the given parameters.
+
+        :keyword simulator_connection: An instance of AzureOpenAIModelConfiguration representing the connection
+            for simulating user response. Defaults to None.
+        :paramtype simulator_connection: Optional[AzureOpenAIModelConfiguration]
+        :keyword ai_client: An instance of AIClient for interacting with the AI service. Defaults to None.
+        :paramtype ai_client: Optional[AIClient]
+        :keyword ml_client: An instance of MLClient for interacting with the AI service. Defaults to None.
+        :paramtype ml_client: Optional[MLClient]
+        :keyword simulate_callback: A callback function that takes a dictionary as input and returns a dictionary.
+            This function is called to simulate the assistant response. Defaults to None.
+        :paramtype simulate_callback: Optional[Callable[[Dict], Dict]]
+
+        :raises ValueError: If `simulator_connection`, `ai_client` and `ml_client` are not provided (i.e., they are None).
+        """
+        if (
+            (ai_client is None and simulator_connection is None and ml_client is None)
+            or (ml_client is not None and simulator_connection is not None)
+            or (ai_client is not None and simulator_connection is not None)
+        ):
+            raise ValueError(
+                "One and only one of the parameters [ai_client, simulator_connection, ml_client] has to be set."
+            )
+
+        if simulate_callback is None:
+            raise ValueError("Callback cannot be None.")
+
+        if not asyncio.iscoroutinefunction(simulate_callback):
+            raise ValueError("Callback has to be an async function.")
+
+        self.ai_client = ai_client
+        self.simulator_connection = self._to_openai_chat_completion_model(simulator_connection)
+        self.adversarial = False
+        self.rai_client = None
+        if ai_client:
+            self.ml_client = ai_client._ml_client
+            self.token_manager = ManagedIdentityAPITokenManager(
+                token_scope=TokenScope.DEFAULT_AZURE_MANAGEMENT,
+                logger=logging.getLogger("managed identity token manager"),
+            )
+            self.rai_client = RAIClient(self.ml_client, self.token_manager)

```

Comment:
Shall we use credentials from ml_client instead of creating new credential class ?